synapse/http/client.py

```diff
@@ -109,7 +111,12 @@ def addressResolved(address):
                     logger.info(
                         "Dropped %s from DNS resolution to %s" % (ip_address, hostname)
                     )
-                    raise SynapseError(403, "IP address blocked by IP blacklist entry")
+                    # Only raise a 403 if this request originated from a
```

Comment:
why do we need to do this?

Reply:
I'm struggling to remember now. I remember I worked with @erikjohnston on this.

Something will passing tests perhaps?

Reply:
T'was actually due to special-casing for unit tests, but we've now removed this, as the existing implementation didn't make sense anyways.